[com_associations] moved "checked out" check to a helper function

rdeutz opened the pull request:
### Summary of Changes
This PR moves the check if an item is checked out to the helper function and used this function when creating links to items

### Testing Instructions
* install latest staging with 2 or more languages 
* create a 2nd user
* open 2 browsers and log in with both users
* browser A: open Multilingual Associations extension
* browser A: chose article and a language
* browser B: open an article (edit)

You will see that you can access an checked out article, this end with an error message but that's not nice.

**Apply the patch**

Now you can see that articles have the locked sign and also the associated items don't have a link to open it.

rdeutz committed:
moved checked our check to a helper function for less gussing

alikon commented:
![multipr](https://cloud.githubusercontent.com/assets/181681/22347609/de1e77f2-e408-11e6-81d4-603450785bd9.PNG)

i've logged in in browser A with admin1
i've logged in in browser B with admin2

after applying the patch on browser B i see the locked icon on article IT and it associated article GB
but both have a link to open is this wanted behaviuor ?

rdeutz commented:
I didn't changed the article view I only worked on the associations view. Seems to me that is as it was before

alikon commented:
sorry  i was confused by 

> items don't have a link to open it.

alikon commented:
I have tested this item :white_check_mark: successfully on 6e67ba17183e165fe126bc0863518bb9b36d154b<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13758">issues.joomla.org/tracker/joomla-cms/13758</a>.</sub>

infograf768 commented:
Not sure it works as should. Maybe need clearer test instructions.

User2 using BrowserB is a manager with access to com_associations.
After patch, here is the page in browser B showing the article which is checked out by another user in browserA.
It does show the lock (greyed as Manager has no com_checkin permission) but the article is still clickable.

![screen shot 2017-01-27 at 10 11 48](https://cloud.githubusercontent.com/assets/869724/22365664/0f340c6a-e479-11e6-8067-a581f3add58a.png)

Now I click on the article title and I get this.

![associationscheckedout3](https://cloud.githubusercontent.com/assets/869724/22366010/bc672588-e47a-11e6-8b3d-1dd392ff23b5.gif)

As you can see, not only I can click on an item which is locked, but also I get weird results in the side by side and when closing and although the user/manager has no com_checkin permissions, the article is unlocked and the associated article (FR in this case) is locked.

infograf768 commented:
I have tested this item :red_circle: unsuccessfully on 6e67ba17183e165fe126bc0863518bb9b36d154b<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13758">issues.joomla.org/tracker/joomla-cms/13758</a>.</sub>

rdeutz commented:
Ok got it. This here moved the check if something is checked out to a helper function. 

The question at hand is, what is the right behaviour? 

a) If we say the article at the left is something like the source than it doesn't makes any sense to edit this or on the associated articles (because some is changing the source). 

b) If this can be source or target than it is much more complicated. 

What we can do is to disable the link on the article in the associations view, when the article is checked out

rdeutz committed:
fix for checked out items

If the item at the left is checked out nothing can be done till the item is checked in

rdeutz commented:
I made a fix following the a) behaviour

infograf768 commented:
OK now: a locked item as well as its associations button (when present) do not open the side by side.

There are other issues to solve but I can mark this as tested OK for what it already does.

infograf768 commented:
I have tested this item :white_check_mark: successfully on 14c6fcbf365d0d06d2d75950ba333b287befeed2<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13758">issues.joomla.org/tracker/joomla-cms/13758</a>.</sub>

infograf768 commented:
@alikon 
Please retest this to move on.

alikon commented:
hope i've not missed some test case this turn ;)

alikon commented:
I have tested this item :white_check_mark: successfully on 14c6fcbf365d0d06d2d75950ba333b287befeed2<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13758">issues.joomla.org/tracker/joomla-cms/13758</a>.</sub>

rdeutz commented:
so we can set RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13758">issues.joomla.org/tracker/joomla-cms/13758</a>.</sub>
